src/otp-input/otp-input.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useEffect, useRef, useState } from "react";
+import { OtpInputProps } from "./types";
+import {
+    CTAButton,
+    ErrorContainer,
+    InputContainer,
+    InputField,
+    Wrapper,
+} from "./otp-input.styles";
+
+export const OtpInput = ({
+    cooldownDuration,
+    ctaProps,
+    errorMessage,
+    numOfInput,
+    onChange,
+    ...otherProps
+}: OtpInputProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const {
+        disabled,
+        onClick,
+        styleType = "secondary",
+        ...otherCtaProps
+    } = ctaProps ?? {};
+    const inputRefs = useRef<(HTMLInputElement | null)[]>([]);
+
+    const [otpValues, setOTPValues] = useState<string[]>(
+        new Array(numOfInput).fill("")
+    );
+    const [countDown, setCountDown] = useState<number>(cooldownDuration);
+    const [enableCountDownTimer, setEnableCountDownTimer] =
+        useState<boolean>(true);
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+    useEffect(() => {
+        inputRefs?.current[0]?.focus();
+        document.addEventListener("paste", handleOnPaste);
+        return () => document.removeEventListener("paste", handleOnPaste);
+    }, []);
+
+    useEffect(() => {
+        if (
+            !enableCountDownTimer ||
+            disabled ||
+            otherProps.disabled ||
+            otherProps.readOnly
+        ) {
+            return;
+        }
+        let timer = cooldownDuration;
+        setCountDown(timer);
+        const interval = setInterval(() => {
+            timer--;
+            setCountDown(timer);
+            if (timer === 0) {
+                clearInterval(interval);
+                setEnableCountDownTimer(false);
+            }
+        }, 1000);
+        return () => clearInterval(interval);
+    }, [enableCountDownTimer]);
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+    const handleOnKeyDown = (
+        event:
+            | React.KeyboardEvent<HTMLInputElement>
+            | React.ChangeEvent<HTMLInputElement>,
+        index: number
+    ) => {
+        const { key } = event as React.KeyboardEvent<HTMLInputElement>;
+        const invalidKeys = ["E", "e", "+", "-", "."];
```

Comment:
Based on past experiences, we had issues trying to track the key properly since some browsers return it as `code` instead. Can I trouble you to test this across browsers to make sure it works. 

Else an alternative is whether you could use the actual value instead? If the value isnt permitted, unset it and dont set it to the `newOtpValues` array. That way, it wouldn't be filled i suppose?

Can refer to [DateInput](https://github.com/LifeSG/react-design-system/blob/8ad923b41805e83832f7a3a52e357324813687b8/src/date-input/date-input.tsx#L195) which has something similar to restrict to numeric values only.

Reply:
to be safe I added both key & code. But yes tested it in Chrome, Safari, Edge & Mozilla.

Reply:
I am able to type `.` using Android Chrome. I still feel the checking of value as I suggested is a more robust approach. Some keys in mobile devices do not translate to an event key or code